Rework bar organization - remove makefiles

ericonr opened the pull request:
Oof, I forgot this was the part you were working on in the other PR, sorry! I can rebase it when you merge that one, no problem.

I think you can get a good idea of what I'm doing here by looking at the commits. The Makefile part is the one that still needs some decisions, because the way I'm doing it is not exactly perfect, since `DATA_DIR_STR` becomes hard coded, while the install `PREFIX` is still configurable.

ericonr commented:
I would like to move some of those clasess to a common folder for all utilities at some point, if you're open to that.

nwg-piotr commented:
> I can rebase it when you merge that one

Merged. I can't do much more while at the office. :/

> I would like to move some of those clasess to a common folder for all utilities at some point, if you're open to that.

Sure. Initially I planned on 3 separate packages, having "bloatware hunters" in mind. This was a mistake from the very beginning.

nwg-piotr commented on bar/bar.cc:
```diff
@@ -41,7 +50,7 @@ int main(int argc, char *argv[]) {
     InputParser input(argc, argv);
     if(input.cmdOptionExists("-h")){
-        std::cout << "GTK button bar: nwgbar " << version << " (c) Piotr Miller 2020\n\n";
+        std::cout << "GTK button bar: nwgbar " << VERSION << " (c) Piotr Miller 2020\n\n";
```
Please replace with ` (c) Piotr Miller & Contributors 2020\n\n` by the way.

ericonr commented:
> Sure. Initially I planned on 3 separate packages, having "bloatware hunters" in mind. This was a mistake from the very beginning.

I think we can still make it possible to have the three separate applications, if one desires.

How would you prefer to deal with the definition strings? Have config take on default values if being run from a makefile?

nwg-piotr commented:
Let's give up on makefiles and separate apps, and Keep It Simple.

ericonr commented:
Ok, updated! It built locally fine, if you could also test it out that would be great!

I added some more stuff to the README, let me know what you think of the changes.

ericonr commented:
Regarding the style of the changes I made, would you like to do them for the other programs yourself or can I open PRs for them later?

nwg-piotr commented:
> would you like to do them for the other programs yourself or can I open PRs for them later?

You'll do it better. If you don't mind, I'll focus on the latest feature request, and some other minor changes.
